### PR TITLE
Private events

### PR DESCRIPTION
**Pull request by John-Arboleda**
This is a site similar to a private [Eventbrite](https://www.eventbrite.com/) which allows users to create events and then manage user signups. Users can create events and send invitations and parties (sound familiar?). Events take place at a specific date and at a location (which you can just store as a string, like “Andy’s House”). A user can create events. A user can attend many events. An event can be attended by many users. 

## Built With

- Ruby on Rails
- Ruby ( version => 2.7.3 )
- yarn
- Bootstrap 5.0

### CODE REVIEW

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
+<h2>Upcoming Events</h2>
+
+<div class="row">
+  <% @events.upcoming.each do |event| %>
```
Comment (cyonii) [inline app/views/events/index.html.erb]: - [ ] Lists upcoming events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    </div>
+  <% end %>
+</div>
+
+<br/>
+
+<h2>Past Events</h2>
+
+<div class="row">
+  <% @events.past.each do |event| %>
```
Comment (cyonii) [inline app/views/events/index.html.erb]: - [ ] Lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <% else %>
+        <% @user.events.each do |event| %>
+          <li><%=link_to event.name, event_path(event) %></li>
+        <% end %>
+      <% end %>
+    </ul>
+  </div>
+
+  <div class="col-md-4">
+    <h3>Upcoming Events: <%= @user.attended_events.where('date >= ?', Date.today).count %></h3>
```
Comment (cyonii) [inline app/views/users/show.html.erb]: - [ ] Lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      <% else %>
+        <% @user.attended_events.where('date >= ?', Date.today).each do |event| %>
+          <li><%=link_to event.name, event_path(event) %></li>
+        <% end %>
+      <% end %>
+    </ul>
+  </div>
+
+  <div class="col-md-4">
+    <h3>Attended Events: <%= @user.attended_events.where('date < ?', Date.today).count %></h3>
```
Comment (cyonii) [inline app/views/users/show.html.erb]: - [ ] Lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by cyonii — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUIRED ♻️✍️

Hello @John-Arboleda, nice project you got there.

#### TO HIGHLIGHT
- Reall great commit messages which shows how the project was built.
- Good and simple README with clear instructions.
- Neat codebase
- Passing linters
- You did particularly well building the MVP, kudos.

#### TO IMPROVE (MANDATORY)
##### Behaviour
- [ ] The event's show page displays the list of attendants  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/events_attendees_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)), currently this is not displayed.
#### Tests
Please make sure to add tests for your project.
- [ ] Models should have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ] Authentication should have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ] Event management features should have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

#### Others
- [ ] Please inline comments


> #### <br>You can ping me on Slack @[CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) to draw my attention in case I do not respond quickly on here. I'm always available and will be glad to help.<br><br>

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:
## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏
# OPTIONAL
- [ ] Views should be completely separate from app logic. You can move view-related logic to app/helpers.
### Happy Coding ☺️
## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Pull request merged**